File: server/router/shop/sku_spec.go
package shop

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type SkuSpecRouter struct{}

// InitSkuSpecRouter 初始化 SKU规格字典 路由信息
func (s *SkuSpecRouter) InitSkuSpecRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	skuSpecRouter := Router.Group("skuSpec").Use(middleware.OperationRecord())
	skuSpecRouterWithoutRecord := Router.Group("skuSpec")
	{
		skuSpecRouter.POST("createSkuSpec", skuSpecApi.CreateSkuSpec)             // 新建SKU规格字典
		skuSpecRouter.DELETE("deleteSkuSpec", skuSpecApi.DeleteSkuSpec)           // 删除SKU规格字典
		skuSpecRouter.DELETE("deleteSkuSpecByIds", skuSpecApi.DeleteSkuSpecByIds) // 批量删除
		skuSpecRouter.PUT("updateSkuSpec", skuSpecApi.UpdateSkuSpec)              // 更新SKU规格字典
	}
	{
		skuSpecRouterWithoutRecord.GET("findSkuSpec", skuSpecApi.FindSkuSpec)       // 根据ID获取
		skuSpecRouterWithoutRecord.GET("getSkuSpecList", skuSpecApi.GetSkuSpecList) // 获取列表
		skuSpecRouterWithoutRecord.GET("getAllSkuSpecs", skuSpecApi.GetAllSkuSpecs) // 获取全部（下拉）
	}
	_ = PublicRouter
}

File: web/src/api/shop/skuSpec.js
import service from '@/utils/request'

/**
 * 创建SKU规格字典
 * @param {Object} data SKU规格字典数据
 * @returns {Promise}
 */
export const createSkuSpec = (data) => {
  return service({
    url: '/skuSpec/createSkuSpec',
    method: 'post',
    data
  })
}

/**
 * 删除SKU规格字典
 * @param {Object} params 删除参数
 * @returns {Promise}
 */
export const deleteSkuSpec = (params) => {
  return service({
    url: '/skuSpec/deleteSkuSpec',
    method: 'delete',
    params
  })
}

/**
 * 批量删除SKU规格字典
 * @param {Object} data 批量删除参数
 * @returns {Promise}
 */
export const deleteSkuSpecByIds = (data) => {
  return service({
    url: '/skuSpec/deleteSkuSpecByIds',
    method: 'delete',
    data
  })
}

/**
 * 更新SKU规格字典
 * @param {Object} data SKU规格字典数据
 * @returns {Promise}
 */
export const updateSkuSpec = (data) => {
  return service({
    url: '/skuSpec/updateSkuSpec',
    method: 'put',
    data
  })
}

/**
 * 根据ID获取SKU规格字典
 * @param {Object} params 查询参数
 * @returns {Promise}
 */
export const findSkuSpec = (params) => {
  return service({
    url: '/skuSpec/findSkuSpec',
    method: 'get',
    params
  })
}

/**
 * 分页获取SKU规格字典列表
 * @param {Object} params 分页查询参数
 * @returns {Promise}
 */
export const getSkuSpecList = (params) => {
  return service({
    url: '/skuSpec/getSkuSpecList',
    method: 'get',
    params
  })
}

/**
 * 获取所有SKU规格字典（下拉选择用）
 * @param {string} type 类型：spec或attr
 * @returns {Promise}
 */
export const getAllSkuSpecs = (type) => {
  return service({
    url: '/skuSpec/getAllSkuSpecs',
    method: 'get',
    params: { type }
  })
}
